guildwars2/account.py

```diff
@@ -297,6 +297,124 @@ def refined_armor_scan(a, b=ids_refined_envoy_armor):
+                    {"name": "Voice in the Void CM", "type": "single", "id": 3979}
+                ],
+                "Fractals": [
+                    {"name": "Nightmare CM", "type": "single", "id": 3180},
+                    {"name": "Shattered Observatory CM", "type": "single", "id": 3465}
+                ],
+        }
+
+        # Create a list of lists of ids.
+        achievement_ids = [[x["id"]] if x["type"] == "single" else x["ids"]
+                for x in chain.from_iterable(wings.values())]
+        # Flatten it.
+        achievement_ids = [str(x) for x in chain.from_iterable(achievement_ids)]
+
+        try:
+            doc = await self.fetch_key(user, scopes)
+            endpoint = "account/achievements?ids=" + ",".join(achievement_ids)
+            results = await self.call_api(endpoint, user, ["progression"])
+        except APINotFound as e:
+            # Not Found is returned by the API when none of the searched
+            # achievements have been completed yet.
+            results = []
+        except APIError as e:
+            return await self.error_handler(ctx, e)
+
+        def is_killed(boss):
+            # If the achievement is finished, the encounter was completed
+            if boss["type"] == "single":
+                for achievement in results:
+                    if achievement["id"] == boss["id"] and achievement["done"]:
+                        return "+";
+                # The achievement is not in the list or isn't done
+                return "-"
+            # If all achievements were completed, the encounter was completed
+            if boss["type"] == "all":
+                for id in boss["ids"]:
+                    found = False
+                    for achievement in results:
+                        if achievement["id"] == id and achievement["done"]:
+                            found = True
+                            break
+                    if not found:
+                        return "-"
+                return "+"
+            # If any of these achievements are completed, the encounter was
+            # completed, otherwise we don't know.
+            if boss["type"] == "any_or_none":
+                for id in boss["ids"]:
+                    for achievement in results:
+                        if achievement["id"] == id and achievement["done"]:
+                            return "+";
+                return "?"
+
+        embed = discord.Embed(title="Kill Proof", color=self.embed_color)
+        embed.set_author(name=doc["account_name"], icon_url=user.avatar_url)
+        for wing in wings:
+            value = ["```diff"]
+            bosses = wings[wing]
+            for boss in bosses:
+                value.append(is_killed(boss) + boss["name"])
+            value.append("```")
+            embed.add_field(name=wing, value="\n".join(value))
+
+        embed.description = "List of encounters that were completed by this player."
```

Comment:
`Kill proof earned` or something along those lines would be better, "by this player" is unnecessary